plugins/org.jboss.tools.vpe.preview.core/src/org/jboss/tools/vpe/preview/core/util/ActionBar.java

```diff
@@ -219,7 +219,9 @@ public boolean isAutomaticRefreshEnabled() {
 	
 	public void updateRefreshItemsAccordingToPreferences() {
 		enableRefreshOnSaveAction.setChecked(preferences.getBoolean(REFRESH_ON_SAVE_PREFERENCES));
+		enableRefreshOnSaveAction.run();
 		enableAutomaticRefreshAction.setChecked(preferences.getBoolean(REFRESH_ON_CHANGE_PREFERENCES));
+		enableAutomaticRefreshAction.run();
```

Comment:
@rawagner so, there was a bug in **updateRefreshItemsAccordingToPreferences** ? Actions were not updated properly without running them heretofore (adding / removing listeners) ?